packages/material-ui/src/styles/withStyles.js

```diff
@@ -210,6 +213,62 @@ const withStyles = (stylesOrCreator, options = {}) => Component => {
       }
     }
 
+    getClasses() {
+      let generate = false;
```

Comment:
`generate` tracks if either the rendered classes or classes prop has changed, requiring the generation of a new finalized classes object.